glotaran/builtin/models/kinetic_image/kinetic_image_dataset_descriptor.py

```diff
@@ -12,11 +12,38 @@
     }
 )
 class KineticImageDatasetDescriptor(DatasetDescriptor):
-    def get_k_matrices(self):
-        return [mat for mat in [cmplx.full_k_matrix() for cmplx in self.megacomplex] if mat]
+    def get_megacomplex_k_matrices(self):
+        scales = (
+            [
+                self.megacomplex_scale[i]
+                for i in [
+                    i
+                    for i, megacomplex in enumerate(self.megacomplex)
+                    if megacomplex.has_k_matrix()
+                ]
+            ]
+            if self.megacomplex_scale is not None
+            else None
+        )
+
+        matrices = [
+            matrix
+            for matrix in [
+                megacomplex.full_k_matrix()
+                for megacomplex in self.megacomplex
+                if megacomplex.has_k_matrix()
+            ]
+        ]
```

Comment:
```suggestion
        matrices = [
                megacomplex.full_k_matrix()
                for megacomplex in self.megacomplex
                if megacomplex.has_k_matrix()
        ]
```
This should do the same but with one less loop.